app/models/course_subject.rb

```diff
@@ -19,14 +21,21 @@ def current_duration
   end
 
   private
+  def add_course_subject_tasks
+    subject.tasks.each do |task|
+      course_subject_tasks.create! name: task.name,
+                                   description: task.description
+    end
```

Comment:
nếu create fail ở đây thì course_subject có rollback không em

Reply:
Có ạ. Nếu create fail ở đây sẽ rollback toàn bộ, vì toàn bộ create dây chuyền này là một commit chứ không tách ra commit riêng ạ.